code/modules/mob/living/carbon/human/species/station.dm

```diff
@@ -378,12 +389,132 @@
+		var/new_body_color = BlendRGB(r_color, rgb(H.r_skin, H.g_skin, H.b_skin), blood_amount/(blood_amount+(H.reagents.total_volume)))
+		var/list/new_color_list = ReadRGB(new_body_color)
+		H.r_skin = new_color_list[1]
+		H.g_skin = new_color_list[2]
+		H.b_skin = new_color_list[3]
+		if(world.time % 200 > 180) 	// Once every 20 seconds - update_body is expensive
+									// The weird time shenanigans are because reagents update only every 2 seconds
+			for(var/organname in H.organs_by_name)
+				var/obj/item/organ/external/E = H.organs_by_name[organname]
+				if(E.dna.species == "Slime People")
+					E.sync_colour_to_human(H)
+			H.regenerate_icons() // Because update_icon didn't work
+	return ..()
+
+/mob/living/carbon/human/proc/regrow_limbs()
+	set category = "IC"
+	set name = "Regrow Limbs"
+	set desc = "Regrow one of your missing limbs at the cost of a large amount of hunger"
+
+	var/const/hungercost = 125
+	var/const/minhunger = 300
+	var/const/regrowthdelay = 450 // 45 seconds
+
+	if(stat || paralysis || stunned || weakened)
+		src << "<span class='warning'>You cannot regenerate missing limbs in your current state.</span>"
+		return
+
+	if(nutrition < minhunger)
+		src << "<span class='warning'>You're too hungry to regenerate a limb!</span>"
+		return
+
+	var/list/missing_limbs = list()
+	for(var/l in organs_by_name)
+		var/obj/item/organ/external/E = organs_by_name[l]
+		if(!istype(E) || istype(E, /obj/item/organ/external/stump))
+			var/list/limblist = species.has_limbs[l]
+			var/obj/item/organ/external/limb = limblist["path"]
+			var/parent_organ = initial(limb.parent_organ)
+			var/obj/item/organ/external/parentLimb = organs_by_name[parent_organ]
+			if(!istype(parentLimb,/obj/item/organ/external) || istype(parentLimb,/obj/item/organ/external/stump))
+				continue
+			missing_limbs[initial(limb.name)] = l
+
+	if(!missing_limbs.len)
+		src << "<span class='warning'>You're not missing any limbs!</span>"
+		return
+
+	var/limb_select = input(src, "Choose a limb to regrow", "Limb Regrowth") as null|anything in missing_limbs
+	var/chosen_limb = missing_limbs[limb_select]
+
+	visible_message("<span class='notice'>[src] begins to hold still and concentrate on their missing [limb_select]...</span>", "<span class='notice'>You begin to focus on regrowing your missing [limb_select]... (This will take [round(regrowthdelay/10)] seconds)</span>")
+	if(do_after(src, regrowthdelay, needhand=0))
+		if(stat || paralysis || stunned || weakened)
+			src << "<span class='warning'>You cannot regenerate missing limbs in your current state.</span>"
+			return
+
+		if(nutrition < minhunger)
+			src << "<span class='warning'>You're too hungry to regenerate a limb!</span>"
+			return
+
+		var/obj/item/organ/external/O = organs_by_name[chosen_limb]
+
+		if(istype(O) && !O.isStump())
+			src << "<span class='warning'>Your limb has already been replaced in some way!</span>"
+			return
+
+		var/stored_brute = 0
+		var/stored_burn = 0
+		if(O.isStump())
```

Comment:
`isStump()` is undefined. Both 485 and 479 are throwing the error `O.isStump: undefined proc`.